application/apps/rustcore/ts-bindings/spec/session.benchmark.spec.ts

```diff
@@ -0,0 +1,306 @@
+// tslint:disable
+
+// We need to provide path to TypeScript types definitions
+/// <reference path="../node_modules/@types/jasmine/index.d.ts" />
+/// <reference path="../node_modules/@types/node/index.d.ts" />
+import { initLogger } from './logger';
+initLogger();
+import { Session, Factory } from '../src/api/session';
+import { IAttachmentsUpdatedUpdated } from '../src/api/session.provider';
+import { IAttachment } from 'platform/types/content';
+import { createSampleFile, finish, performanceReport, setMeasurement, runner } from './common';
+import { readBenchmarkConfigurationFile } from './config_benchmarks';
+import { IndexingMode } from 'platform/types/content';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+const config = readBenchmarkConfigurationFile().get().tests.benchmark;
+
+describe('Benchmark Tests', function () {
+    Object.keys(config.performance.tests).forEach((alias: string) => {
+        const test = config.performance.tests[alias];
+        const testName = `${test.alias}`;
+        if (test.ignore) {
+            console.log(`Test "${testName}" has been ignored`);
+            return;
+        }
+
+        it(testName, function () {
+            return runner(
+                    {
+                        open_as: '',
+                        ignore: false,
+                        alias: testName,
+                        expectation_ms: 10000,
+                        file: '',
+                    },
+                    1,
+                    async (logger, done, collector) => {
+                    const measurement = setMeasurement();
+                    try {
+                        const session = await Session.create();
+                        session.debug(true, testName);
+
+                        const stream = session.getStream();
+                        if (stream instanceof Error) throw stream;
+
+                        const events = session.getEvents();
+                        if (events instanceof Error) throw events;
+
+                        const search = session.getSearch();
+
+                        let home_dir = (process.env as any)['SH_HOME_DIR'];
```

Comment:
Potential bug. Case if `home_dir` doesn't have the correct value (at least not an empty string) isn't handled